packages/protocol/test/stability/stabletokenregistry.ts

```diff
@@ -0,0 +1,157 @@
+import { assertRevert } from '@celo/protocol/lib/test-utils'
+import BigNumber from 'bignumber.js'
+import { StableTokenRegistryContract, StableTokenRegistryInstance } from 'types'
+
+const STRC: StableTokenRegistryContract = artifacts.require('StableTokenRegistry')
+
+const convertToHex = (input: string) => {
+  return web3.utils.utf8ToHex(input)
+}
+
+const assertSTContractNames = (
+  contractsHex: string,
+  lengths: BigNumber[],
+  expectedContracts: string[]
+) => {
+  assert.equal(lengths.length, expectedContracts.length)
+  const contracts = web3.utils.hexToUtf8(contractsHex)
+  let currentIndex = 0
+  expectedContracts.forEach((expectedContract: string, i: number) => {
+    const contract = contracts.slice(currentIndex, currentIndex + lengths[i].toNumber())
+    currentIndex += lengths[i].toNumber()
+    assert.equal(contract, expectedContract)
+  })
+  assert.equal(contracts.length, currentIndex)
+}
+
+contract('StableTokenRegistry', (accounts: string[]) => {
+  let strc: StableTokenRegistryInstance
+  const nonOwner: string = accounts[1]
+
+  const fiatTicker: string = convertToHex('USD')
+  const stableTokenContractName = convertToHex('StableToken')
+
+  const getFiatTickers = async () => {
+    const updatedFiatTickers = []
+    try {
+      let index = 0
+      while (await strc.fiatTickers(index)) {
+        updatedFiatTickers.push(web3.utils.hexToUtf8(await strc.fiatTickers(index)))
+        index++
+      }
+    } catch (error) {
+      return updatedFiatTickers
+    }
+  }
+
+  beforeEach(async () => {
+    strc = await STRC.new(true)
+    await strc.initialize(convertToHex('GEL'), convertToHex('StableTokenGEL'))
+  })
+
+  describe('#initialize()', async () => {
+    it('should have set the owner', async () => {
+      const owner: string = await strc.owner()
+      assert.equal(owner, accounts[0])
+    })
+
+    it('should not be callable again', async () => {
+      await assertRevert(strc.initialize(convertToHex('GEL'), convertToHex('StableTokenGEL')))
+    })
+  })
+
+  describe('#removeStableToken(fiatTicker)', () => {
+    beforeEach(async () => {
+      await strc.addNewStableToken(convertToHex('GBP'), convertToHex('StableTokenGBP'))
+    })
+
+    it('only allows owner', async () => {
+      await assertRevert(strc.removeStableToken(fiatTicker, 0, { from: nonOwner }))
+    })
+
+    it('has the right list of fiat tickers after removing one', async () => {
+      await strc.removeStableToken(fiatTicker, 0)
+      const updatedFiatTickers = await getFiatTickers()
```

Comment:
The registry should be passed as a parameter here rather than having it implicitly

Reply:
@martinvol not sure I understand what you mean here

Reply:
Like it should be explicit `await getFiatTickers(strc)`, otherwise it is not trivial what things are you comparing

Reply:
`getFiatTickers() ` is just a helper implemented [here](https://github.com/celo-org/celo-monorepo/blob/677ada225a452b65b94a5a0c7ffcc37f7951e819/packages/protocol/test/stability/stabletokenregistry.ts#L34), because this is just decoding the arb byte data off-chain